src/AspNetCoreModuleV2/CommonLib/hostfxroptions.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "stdafx.h"
+
+HRESULT HOSTFXR_OPTIONS::Create(
+        _In_ PCWSTR         pcwzExeLocation,
+        _In_ PCWSTR         pcwzProcessPath,
+        _In_ PCWSTR         pcwzApplicationPhysicalPath,
+        _In_ PCWSTR         pcwzArguments,
+        _In_ HANDLE         hEventLog,
+        _Out_ std::unique_ptr<HOSTFXR_OPTIONS>& ppWrapper)
+{
+    HRESULT hr = S_OK;
+    STRU struHostFxrDllLocation;
+    STRU struExeAbsolutePath;
+    STRU struExeLocation;
+    BSTR* pwzArgv;
+    DWORD dwArgCount;
+
+    if (pcwzExeLocation != NULL && FAILED(struExeLocation.Copy(pcwzExeLocation)))
+    {
+        goto Finished;
+    }
+
+    // If the exe was not provided by the shim, reobtain the hostfxr parameters (which finds dotnet).
+    if (struExeLocation.IsEmpty())
+    {
+        if (FAILED(hr = HOSTFXR_UTILITY::GetHostFxrParameters(
+            hEventLog,
+            pcwzProcessPath,
+            pcwzApplicationPhysicalPath,
+            pcwzArguments,
+            &struHostFxrDllLocation,
+            &struExeAbsolutePath,
+            &dwArgCount,
+            &pwzArgv)))
+        {
+            goto Finished;
+        }
+    }
+    else if (HOSTFXR_UTILITY::IsDotnetExecutable(&struExeLocation))
+    {
+        if (FAILED(hr = HOSTFXR_UTILITY::ParseHostfxrArguments(
+            pcwzArguments,
+            pcwzExeLocation,
+            pcwzApplicationPhysicalPath,
+            hEventLog,
+            &dwArgCount,
+            &pwzArgv)))
+        {
+            goto Finished;
+        }
+    }
+    else
+    {
+        if (FAILED(hr = HOSTFXR_UTILITY::GetStandaloneHostfxrParameters(
+            pcwzExeLocation,
+            pcwzApplicationPhysicalPath,
+            pcwzArguments,
+            hEventLog,
+            &struHostFxrDllLocation,
+            &dwArgCount,
+            &pwzArgv)))
+        {
+            goto Finished;
+        }
+    }
+
+    ppWrapper = std::make_unique<HOSTFXR_OPTIONS>();
+    if (FAILED(hr = ppWrapper->Populate(struHostFxrDllLocation.QueryStr(), struExeAbsolutePath.QueryStr(), dwArgCount, pwzArgv)))
+    {
+        goto Finished;
+    }
+
+Finished:
+
+    return hr;
+}
+
+
+HRESULT HOSTFXR_OPTIONS::Populate(PCWSTR hostFxrLocation, PCWSTR struExeLocation, DWORD argc, BSTR argv[])
```

Comment:
hostfxr_main accepts WCHAR* for argv: https://github.com/dotnet/core-setup/blob/master/src/corehost/common/pal.h#L103. Let's file an issue to fix it.

Reply:
what do you plan to fix? This function does not guard the scenario that the  input of argc is larger than the size of argv, even though it is a private function